nexus/auth/src/authz/api_resources.rs

```diff
@@ -103,27 +101,27 @@ pub trait ApiResourceWithRoles: ApiResource {
 pub trait ApiResourceWithRolesType: ApiResourceWithRoles {
     type AllowedRoles: serde::Serialize
         + serde::de::DeserializeOwned
-        + db::model::DatabaseString
+        + nexus_db_model::DatabaseString
         + Clone;
 }
 
-impl<T: ApiResource + oso::PolarClass + Clone> AuthorizedResource for T {
-    fn load_roles<'a, 'b, 'c, 'd, 'e, 'f>(
+impl<T> AuthorizedResource for T
+where
+    T: ApiResource + oso::PolarClass + Clone,
+{
+    fn load_roles<'a, 'b, 'c, 'd, 'e>(
         &'a self,
         opctx: &'b OpContext,
-        datastore: &'c DataStore,
-        authn: &'d authn::Context,
-        roleset: &'e mut RoleSet,
-    ) -> BoxFuture<'f, Result<(), Error>>
+        authn: &'c authn::Context,
+        roleset: &'d mut RoleSet,
+    ) -> BoxFuture<'e, Result<(), Error>>
     where
-        'a: 'f,
-        'b: 'f,
-        'c: 'f,
-        'd: 'f,
-        'e: 'f,
+        'a: 'e,
+        'b: 'e,
+        'c: 'e,
+        'd: 'e,
```

Comment:
Nit, take it or leave it: personally, I'd prefer to give these lifetimes semantically meaningful names, like
```rust
    fn load_roles<'resource, 'opctx, 'authn, 'roles, 'future>(
        &'resource self,
        opctx: &'opctx OpContext,
        authn: &'authn authn::Context,
        roleset: &' mut RoleSet,
    ) -> futures::future::BoxFuture<'future, Result<(), Error>>
    where
        'resource: 'future,
        'opctx: 'future,
        'authn 'future,
        'roleset: 'future,
    {
```

because, honestly, seeing
```rust
    where
        'a: 'e,
        'b: 'e,
        'c: 'e,
        'd: 'e,
```
just kinda feels like nonsense to me.

Up to you, though; I agree that single-letter lifetime param names does seem to be the more common idiom... :woman_shrugging:

(as an aside, I'm not convinced we even need so many lifetime params --- can't they all just be `'future` references, and the compiler will coerce them? but maybe that doesn't actually work, I dunno...)

Reply:
(this comment also applies, you know...everywhere else)

Reply:
Completely agree with you on both accounts. I'll try to get rid of these, seems a little surprising that they ever existed in the first place.

Reply:
Okay, I have:

- Reduced this to a single lifetime parameter
- Renamed it `'fut`, to make it more clear "this is here for the BoxFuture return value"

Reply:
oh, that's _way_ better! thanks! :)